src/lib/storage/dictionary_segment/attribute_vector_iterable.hpp

```diff
@@ -51,8 +50,8 @@ class AttributeVectorIterable : public PointAccessibleSegmentIterable<AttributeV
   class Iterator : public BaseSegmentIterator<Iterator<ZsIteratorType>, SegmentPosition<ValueID>> {
    public:
     using ValueType = ValueID;
-    explicit Iterator(const ValueID null_value_id, ZsIteratorType attribute_it, ChunkOffset chunk_offset)
-        : _null_value_id{null_value_id}, _attribute_it{attribute_it}, _chunk_offset{chunk_offset} {}
+    explicit Iterator(const ValueID null_value_id, ZsIteratorType&& attribute_it, ChunkOffset chunk_offset)
```

Comment:
Why not by value?

Reply:
Same in the rest of this file

Reply:
Tbh, I merely copied this parameter handling from the move constructors when I saw that we create iterator copies in the segments iterator constructors.

I passing by value and using `std::move` in the initializer list better here?

Reply:
What I was wondering was: when somebody passes an lvalue and I move that, that would corrupt his data, wouldn't it? For my understanding, this wouldn't work with an rref.

Reply:
https://stackoverflow.com/a/52276689/2204581

If you accept by value, it is not the caller's data anymore, but a copy just made for you. It is thus the caller responsibility to move it the iterator if it wants to relinquish control of it.

Reply:
Yes, but I’d like to rule out that any iterators are copied because somebody did not use `std::move`.

Reply:
ok